qt/scientific_interfaces/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1059,19 +1063,21 @@ void EnggDiffractionPresenter::doCalib(const EnggDiffCalibSettings &cs,
                                        const std::string &ceriaNo,
                                        const std::string &outFilename,
                                        const std::string &specNos) {
-  ITableWorkspace_sptr vanIntegWS;
-  MatrixWorkspace_sptr vanCurvesWS;
   MatrixWorkspace_sptr ceriaWS;
 
   // Append current instrument name if numerical only entry
   // to help Load algorithm determine instrument
   std::string vanFileHint, cerFileHint;
   appendCalibInstPrefix(vanNo, ceriaNo, vanFileHint, cerFileHint);
 
-  // save vanIntegWS and vanCurvesWS as open genie
-  // see where spec number comes from
-  loadOrCalcVanadiumWorkspaces(vanFileHint, cs.m_inputDirCalib, vanIntegWS,
-                               vanCurvesWS, cs.m_forceRecalcOverwrite, specNos);
+  // TODO: the settings tab should emit a signal when these are changed, on
+  // which the vanadium corrections model should be updated automatically
+  m_vanadiumCorrectionsModel->setCalibSettings(cs);
+  m_vanadiumCorrectionsModel->setCurrentInstrument(m_view->currentInstrument());
+  const auto vanadiumCorrectionWorkspaces =
+      m_vanadiumCorrectionsModel->fetchCorrectionWorkspaces(vanNo);
+  const auto vanIntegWS = vanadiumCorrectionWorkspaces.first;
+  const auto vanCurvesWS = vanadiumCorrectionWorkspaces.second;
```

Comment:
Not that it matters much performance-wise, but as far as I can see, `vanIntegWS` and `vanCurvesWS` could be references avoiding copying the shared pointers.